.github/workflows/main.yaml

```
@@ -31,28 +31,16 @@ jobs:
           python-version: ${{ matrix.python-version }}
           cache: "pip"
           cache-dependency-path: |
-            **/setup.cfg
+            **/pyproject.toml
             **/requirements*.txt
             **/test-requirements*.txt
 
       - name: Install dependencies
         run: pip install -r test-requirements.txt --upgrade pip
 
       - if: matrix.python-version == '3.10'
```

```suggestion
      - if: matrix.python-version == '3.10'
```

Should we make it so that only testing happens on 3.10 and the linting and such run on latest (3.12 in this case?)

---

Same for uploading to Codecov below

---

👋 Thanks for your review!

> Should we make it so that only testing happens on 3.10 and the linting and such run on latest (3.12 in this case?)

It's probably best for us to keep any linters targeting `3.10`, as it's our minimum supported target. Running with later versions could potentially result in linter suggestions that leverage language features unavailable in older versions.

In terms of testing, I think there are two significant advantages to running them against our full gauntlet of supported runtimes: avoiding incorporating language features that aren't supported across all our targets and catching potential compatibility issues with dependencies that could be installed differently between those various runtimes.

> Same for uploading to Codecov and releasing below

For releasing, it's best to build in an environment configured for the minimum supported runtime version just to guarantee the built package will work as intended.